include/zephyr/drivers/misc/pio_rpi_pico/pio_rpi_pico.h

```diff
@@ -148,7 +148,10 @@
  * @param dev Pointer to device structure for rpi_pio device instance
  * @return PIO object
  */
-PIO pio_rpi_pico_get_pio(const struct device *dev);
+inline PIO pio_rpi_pico_get_pio(const struct device *dev)
+{
+	return *(PIO *)(dev->config);
```

Comment:
It's better to use `CONTAINER_OF`.

Reply:
Hmmmm... I like this idea, but doing so will have to expose `struct pio_rpi_pico_config`, which is private to `pio_rpi_pico.c` (I tried to move this definition to `pio_rpi_pico.h` since it will also expose the clock and reset).

Reply:
Got it. Let's go ahead and do it.